AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1580,15 +1567,25 @@ - (void)measureNodeWithBoundsIfNecessary:(CGRect)bounds
   }
 }
 
-- (ASSizeRange)conrainedSizeForCurrentLayout
+- (ASSizeRange)constrainedSizeForLayoutPass
 {
-  ASSizeRange constrainedSize = ASSizeRangeMake(self.bounds.size);
-
-  // Checkout if constrained size of layouts can be reused
-  if (_pendingDisplayNodeLayout != nullptr) {
+  // Use as default constrained size the bounds
+  CGRect bounds = self.threadSafeBounds;
+  ASSizeRange constrainedSize = ASSizeRangeMake(bounds.size);
+  
+  // Checkout if constrained size of pending or calculated display node layout can be used
+  if (_pendingDisplayNodeLayout != nullptr && CGSizeEqualToSize(_pendingDisplayNodeLayout->layout.size, bounds.size)) {
```

Comment:
@maicki I think it's important that we _not_ check the layout.size == bounds.size.  This is not necessary, as long as _pendingDisplayNodeLayout exists.  Even if _pending isValid == NO, we should still use the constrained size.

The idea is the parent must ACTIVELY inform us if the constrained size changes.  Until such an active change from the parent, this represents our most recently known / current / up-to-date constrained size.